nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayTryFinallyCases.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.testdata;
+
+import javax.annotation.Nullable;
+
+// This code tests our, admitedly quite unsound, handling of try{ ... }finally{ ... } blocks.
+public class NullAwayTryFinallyCases {
+
+  public void derefOnFinallySafe(@Nullable Object o) {
+    try {
+      if (o == null) {
+        return;
+      }
+    } finally {
+      System.out.println(o.toString());
```

Comment:
How is this one safe?  Doesn't it throw an NPE if you pass in null?

Reply:
It absolutely isn't! Apparently both I and NullAway/CF were assuming `finally` doesn't execute after `return` (but does after `throw` or as a simple fall through). This is a bug (see the newly added `tryFinallyMininalRet` for a minimal testcase). Will add a few more cases (and rename this one), then start fixing the inconsistent behavior.

Reply:
Yeah I think this is probably a bug in the CFG that we should report upstream.

Reply:
Maybe the next step is to reproduce some of the weird behaviors in this file (all the `ToDo` comments) on top the CF Nullness checker. Easier for giving them repro cases.